public_desktop/desktop-app.js

```diff
@@ -90,8 +92,60 @@ function startDesktopApp() {
 		}
 	}
 
+	function contextMenu( ev ) {
```

Comment:
Something for future: we could probably split this up into smaller functions.

- `handleContextMenu` or `contextMenuCallback` that determines which menu to show.
- `showEditorContextMenu` and `showGeneralContextMenu` which generate the menu and trigger it.
- `triggerContentMenu` which has the `setTimeout` call.

Once broken up, we can have the `mce-contextmenu` callback just trigger `showEditorContextMenu` (and would help avoid any issues with event objects coming from different contexts).